### PR TITLE
Make autogold tests more readable

### PR DESCRIPTION
**Pull request by camdencheek**
Stacked on #34595

This is a rather opinionated change, so feel free to speak up if you prefer the old style.

Basically, I really struggle to read, use, and update autogold tests when written in the current style.
1) The files are massive, meaning I have to do a lot of scrolling to see what cases are being tested. Additionally, the big multi-line strings make it tough to grok since it breaks my mental indentation model.
2) The tests read backwards. The expected output comes before the input. This means I often read the test name, then scroll down to the test input, then scroll back up to the test output. 
3) When a test fails, it's difficult to tell which one is failing because the test names aren't useful.
4) When tests change, I can't autoupdate all tests at the same time. I have to do something like `while true ; do go test ./package -update ; done`

This changes the tests in a couple of ways to address these. 
- All test cases have a name and an input string, and all test inputs are in a single table.
- Each case is run as a separate sub test. This fixes both (3) and (4) because each subtest has a different name, and autogold can update all the subtests in one `-update` run.
- For tests with large, multiline outputs (i.e. the updated test), the output contents are moved into separate test files, taking advantage of `autogold.Equal`. 

The disadvantage is, for large test outputs like this test, the "expected" value is not in-file. 

Writing this all up because I'd like to propose this as a better autogold pattern moving forward.

## Test plan

I ran the tests.

### CODE REVIEW

File: internal/search/job/jobutil/job_test.go [hunk 1]
```diff
@@ -214,362 +214,41 @@ func Test_optimizeJobs(t *testing.T) {
 
 		baseJob, _ := NewJob(inputs, plan, IdentityPass)
 		optimizedJob, _ := NewJob(inputs, plan, OptimizationPass)
-		return "\nBASE:\n\n" + PrettySexp(baseJob) + "\n\nOPTIMIZED:\n\n" + PrettySexp(optimizedJob) + "\n"
+		return "INPUT:\n\n" + input + "\n\nBASE:\n\n" + PrettySexp(baseJob) + "\n\nOPTIMIZED:\n\n" + PrettySexp(optimizedJob) + "\n"
```
Comment (camdencheek) [inline internal/search/job/jobutil/job_test.go:217]: Note: I added the input to the output string in this case so it's easier to just look at the generated test files.

**Review by rvantonder — APPROVED** [review]
> The files are massive, meaning I have to do a lot of scrolling to see what cases are being tested.

Yeah, good reason for these tests to use files. They started small but got larger as we added info.

> The tests read backwards.

Agree this is troublesome. Unfortunate limitation of autogold that's out of my control. Could be argued I'm not structuring the tests the way autogold wants, but the way autogold wants also diverges from other inline expect/snapshot tests. So I just agree that yes this isn't great.

> I can't autoupdate all tests at the same time.

Yeah big limitation that's out of my control until I go dig into that code :P 

I'm fine with this table-based approach, etc. My big take about inline/expect test changes is: if I'm spending _a lot_ of time reading or writing test code there's something wrong. So I think I feel less annoyed by (1), (2), (3) than you probably do (but still mildly inconvenienced, to some degree). My flow, when I, say, iterate on this file or tests, I mostly just want to know _when_ something breaks, and I can look at the diff if necessary (whichever one I find first in my terminal output) and often that's pretty quick.

What I'm trying to say is: I think this is a good change and I support it! I think the delta of "this is an improvement" is probably larger for you than it is for me, and the delta here is smaller than the threshold for me to go in and improve it :P But I think it's an objectively better improvement for everyone everywhere.

**Review by rvantonder — COMMENTED** [review]
Oh, can we not create gold files with `(` and `)` in them please? Let's not port those into the file names.

**Comment by camdencheek** [comment]
> the delta here is smaller than the threshold for me to go in and improve it

Yeah, I don't expect us to take the time to go spend cycles on test readability, but as I'm in updating tests, I'll probably move in this direction. Thanks for the thoughts 🙂

**Pull request merged**